### PR TITLE
Enable multi-model

### PR DESCRIPTION
**Pull request by tgerdesnv**
Enable multi-model (--run-config-profile-models-concurrently-enable)
Added L0_multi_model_profile test
Updated PA get_cmd() for multi-model to use MPI
Updated auto_adjust_parameters() support to handle multi-model

### CODE REVIEW

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 1]
```diff
@@ -263,8 +304,11 @@ def _poll_perf_analyzer(self, process):
 
     def _auto_adjust_parameters(self, process):
         """
-        Use of the perf analyzer process
+        Attempt to update PA parameters based on the output
         """
+        # TODO-TMA-518 - how to handle?
```
Comment (tgerdesnv) [inline model_analyzer/perf_analyzer/perf_analyzer.py:309]: @matthewkotila I need to figure out how to handle this auto-adjusting for multi-model.

File: model_analyzer/record/metrics_manager.py [hunk 2]
```diff
@@ -439,7 +439,7 @@ def _run_perf_analyzer(self, run_config, perf_output_writer):
         status = perf_analyzer.run(self._perf_metrics, env=perf_analyzer_env)
 
         if perf_output_writer:
-            # TODO-TMA-518: MPI command
+            # TODO-TMA-518: MPI command - still need to fix
```
Comment (tgerdesnv) [inline model_analyzer/record/metrics_manager.py:442]: Random printing outside of PA needs to be cleaned up

File: model_analyzer/perf_analyzer/perf_analyzer.py (hunk 1 above)
Comment (tgerdesnv) [reply model_analyzer/perf_analyzer/perf_analyzer.py]: fixed

File: model_analyzer/perf_analyzer/perf_analyzer.py [hunk 3]
```diff
@@ -104,7 +104,7 @@ def __init__(self, path, config, max_retries, timeout, max_cpu_util):
         self._perf_records = {}
         self._max_cpu_util = max_cpu_util
 
-        # TODO-TMA-518: Need to update for multi-model
+        # TODO-TMA-518: Need to update for multi-model - still need to fix
```
[the remark this answers is not in the document]
Comment (tgerdesnv) [reply model_analyzer/perf_analyzer/perf_analyzer.py]: fixed

File: model_analyzer/record/metrics_manager.py (hunk 2 above)
Comment (tgerdesnv) [reply model_analyzer/record/metrics_manager.py]: fixed

File: qa/L0_multi_model_profile/check_results.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        """
+        Check that each model was profiled the number of times
+        corresponding with batch size and concurrency combinations
+
+        For multi-model with 2 models, the expected number of measurements
+        is X*Y, where:
+        - X is the number of measurements that we would expected 
+        for model 1 in a single-model case
+        - Y is the number of measurements that we would expected 
+        for model 1 in a single-model case
```
Comment (nv-braf) [inline qa/L0_multi_model_profile/check_results.py:50]: this should be model 2 (or you can change this to model A/B which is my preference: models get letters, configs get numbers)

Comment (tgerdesnv) [reply qa/L0_multi_model_profile/check_results.py]: fixed

**Review by nv-braf — APPROVED** [review]

**Review by nv-braf — CHANGES_REQUESTED** [review]
I've observed a crash with my local run - we need to debug before this is released

**Comment by tgerdesnv** [comment]
> I've observed a crash with my local run - we need to debug before this is released

Fixed! I also fixed two other issues -- one a potential issue around unsorted batch size list

**Review by nv-braf — APPROVED** [review]
Looks good!

**Pull request merged**